docs/source/contribution.rst

```diff
@@ -443,3 +291,110 @@ Open ``index.html`` with the browser and see if it is rendered as expected.
+While the major, minor, and revision numbers follow the rule of semantic versioning, the pre-release suffix follows PEP 440, the Python community standards.
+
+**Note that a major update basically does not contain compatibility-breaking changes from the last release candidate (RC).**
+This is not a strict rule, though; if there is a critical bug in the API that need to be fixed for the major version, breaking changes may be introduced.
+
+For more on backward compatibility, please refer to the :ref:`compatibility`.
+
+.. _contrib-release-cycle:
+
+Release Cycle
+~~~~~~~~~~~~~
+
+Two tracks with different versions are developed in parallel.
+The first track is the **stable versions**, which is a series of revision updates for the latest major version.
+The second track is the **development versions**, which is a series of pre-releases for the upcoming major version.
+
+If ``X.0.0`` is the latest major version, followed by ``Y.0.0`` and ``Z.0.0``, the timeline with updates becomes as follows.
+
+========== =========== =========== ============
+   Date       ver X       ver Y       ver Z
+========== =========== =========== ============
+  0 weeks    X.0.0rc1    --         --
+  4 weeks    X.0.0       Y.0.0a1    --
+  8 weeks    X.1.0*      Y.0.0b1    --
+ 12 weeks    X.2.0*      Y.0.0rc1   --
+ 16 weeks    --          Y.0.0      Z.0.0a1
+========== =========== =========== ============
+
+(* These might be revision releases)
+
+The dates shown in the left-most column are relative to the release of ``X.0.0rc1``.
+In particular, each revision/minor release is made four weeks after the previous one of the same major version, and the pre-release of the upcoming major version is made at the same time.
+Whether these releases are revision or minor is determined based on the contents of each update.
+
+Note that there are only three stable releases for the versions ``X.x.x``.
+During the parallel development of ``Y.0.0`` and ``Z.0.0a1``, the version ``Y`` is treated as an **almost-stable version** and ``Z`` is treated as a development version.
+
+If there is a critical bug found in ``X.x.x`` after stopping the development of version ``X``, we may release a hot-fix for this version at any time.
+
+A `milestone for each upcoming release is published on GitHub <https://github.com/chainer/chainer/milestones>`_.
+The GitHub milestones are used to group issues and PRs belonging to a release.
+
+.. _contrib-git-branches:
+
+Git Branches
+~~~~~~~~~~~~
+
+The ``master`` branch is used to develop pre-release versions.
+It means that **alpha, beta, and RC updates are developed at the** ``master`` **branch**.
+This branch contains the most up-to-date source tree that includes features newly added after the latest major version.
+
+The stable version is developed on the ``vN`` branch where "N" reflects the version number (*versioned branch*).
+For example, v3.0.0, v3.0.1, and v3.0.2 are developed on the ``v3`` branch.
```

Comment:
Better to use minor version numbers since they are more realistic for now.

Reply:
Fixed.